ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala

```diff
@@ -57,27 +57,42 @@ object HttpService extends StrictLogging {
 
   final case class Error(message: String)
 
+  // defined separately from Config so
+  //  1. it is absolutely lexically apparent what `import startSettings._` means
+  //  2. avoid incorporating other Config'd things into "the shared args to start"
+  trait StartSettings {
+    val ledgerHost: String
+    val ledgerPort: Int
+    val applicationId: ApplicationId
+    val address: String
+    val httpPort: Int
+    val portFile: Option[Path]
+    val tlsConfig: TlsConfiguration
+    val wsConfig: Option[WebsocketConfig]
+    val accessTokenFile: Option[Path]
+    val allowNonHttps: Boolean
+    val staticContentConfig: Option[StaticContentConfig]
+    val packageReloadInterval: FiniteDuration
+    val maxInboundMessageSize: Int
+  }
+
+  trait DefaultStartSettings extends StartSettings {
+    override val staticContentConfig: Option[StaticContentConfig] = None
+    override val packageReloadInterval: FiniteDuration = DefaultPackageReloadInterval
+    override val maxInboundMessageSize: Int = DefaultMaxInboundMessageSize
+  }
+
   def start(
-      ledgerHost: String,
-      ledgerPort: Int,
-      applicationId: ApplicationId,
-      address: String,
-      httpPort: Int,
-      portFile: Option[Path],
-      tlsConfig: TlsConfiguration,
-      wsConfig: Option[WebsocketConfig],
-      accessTokenFile: Option[Path],
+      startSettings: StartSettings,
       contractDao: Option[ContractDao] = None,
-      staticContentConfig: Option[StaticContentConfig] = None,
-      packageReloadInterval: FiniteDuration = DefaultPackageReloadInterval,
-      maxInboundMessageSize: Int = DefaultMaxInboundMessageSize,
```

Comment:
@leo-da This signature was starting to bug me so I made it all type-overlappy with `Config`, with the "argument" part still being publicly accessible.

Reply:
:+1: